browser/fragments/platform-nativescript.js

```diff
@@ -0,0 +1,71 @@
+require('nativescript-websockets');
+
+/**
+ * atob polyfill
+ * @param input
+ * @returns {string}
+ * Based on From: https://github.com/davidchambers/Base64.js/blob/master/base64.js
+ * License is Apache 2.0 also
+ */
+global.atob = function(input) {
+        var chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+        var str = String(input).replace(/=+$/, '');
+        if (str.length % 4 !== 0) {
+            throw new Error("'atob' failed: The string to be decoded is not correctly encoded.");
+        }
+        for (
+            // initialize result and counters
+            var bc = 0, bs, buffer, idx = 0, output = '';
+            // get next character
+            buffer = str.charAt(idx++);
+            // character found in table? initialize bit storage and add its ascii value;
+            ~buffer && (bs = bc % 4 ? bs * 64 + buffer : buffer,
+                // and if not first of each 4 characters,
+                // convert the first 8 bits to one ascii character
+            bc++ % 4) ? output += String.fromCharCode(255 & bs >> (-2 * bc & 6)) : 0
+        ) {
+            // try to find character in table (0-63, not found => -1)
+            buffer = chars.indexOf(buffer);
+        }
+        return output;
+    };
+
+var randomBytes;
+if (global.android) {
+    randomBytes = function(size) {
+        var sr = new java.security.SecureRandom();
+        var buffer = java.lang.reflect.Array.newInstance(java.lang.Byte.class.getField("TYPE").get(null), size);
```

Comment:
Isn't there now a more efficient way of doing this? (https://github.com/NativeScript/android-runtime/issues/70)